kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -387,6 +409,15 @@ private Protocol getInitialProtocol() {
     return new Protocol(DEFAULT_READ_VERSION, DEFAULT_WRITE_VERSION);
   }
 
+  private void addSystemDomainMetadata() {
```

Comment:
@allisonport-db IIRC, you mentioned that we need a separate API for setting system-controlled domain?

Reply:
This is going to conflict with https://github.com/delta-io/delta/pull/4245 by the way.

I don't think we need a separate API. We probably should pass the `clusteringColumns` through to the txn, and then in the txn we add any internals system DM (using `txnImpl.addDomainMetadataInternal`)